lms/static/scripts/frontend_apps/components/URLFormWithPreview.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  ArrowRightIcon,
+  CancelIcon,
+  IconButton,
+  Input,
+  InputGroup,
+  Thumbnail,
+} from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import type { ComponentChildren, RefObject } from 'preact';
+
+import { useUniqueId } from '../utils/hooks';
+
+export type ThumbnailData = {
+  image?: string;
+  alt?: string;
+  isLoading?: boolean;
+};
+
+export type URLFormWithPreviewProps = {
+  children?: ComponentChildren;
+  thumbnail?: ThumbnailData;
+  error?: string;
+  inputRef: RefObject<HTMLInputElement | undefined>;
+  urlPlaceholder?: string;
+  onURLChange: (inputUrl: string) => void;
+  label: string;
+  defaultURL?: string;
+};
+
+export default function URLFormWithPreview({
+  children,
+  thumbnail,
+  error,
+  inputRef,
+  urlPlaceholder,
+  onURLChange,
+  label,
+  defaultURL,
+}: URLFormWithPreviewProps) {
+  const inputId = useUniqueId('url');
+  const onChange = () => onURLChange(inputRef.current!.value);
+  const onKeyDown = (event: KeyboardEvent) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      event.stopPropagation();
+      onChange();
+    }
+  };
+
+  return (
+    <div className="flex flex-row space-x-2">
+      <div
+        className={classnames(
+          // Negative vertical margins allow thumbnail to use up more vertical
+          // space in the containing Modal
+          '-mb-12 -mt-2 w-[200px] min-w-[200px] h-[200px]'
```

Comment:
I still need to tweak this for video thumbnails specifically, which are usually landscapes, but I want to experiment with different options.